Source/Core/AudioCommon/WaveFile.cpp

```diff
@@ -38,6 +38,15 @@ bool WaveFileWriter::Start(const std::string& filename, unsigned int HLESampleRa
 
 	audio_size = 0;
 
+	if (basename.empty())
+	{
+		std::string filepath = File::GetUserPath(D_DUMPAUDIO_IDX);
+		basename = filename;
+		basename.replace(0, filepath.length(), "");
+		basename.replace(basename.end() - 4, basename.end(), "");
+	}
+	current_sample_rate = HLESampleRate;
```

Comment:
May you start with current_sample_rate=0 and open a new file on the first audio push? So you would also catch the silence on bootup.